test/extended/apiserver/disruption.go

```diff
@@ -0,0 +1,496 @@
+package apiserver
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os/exec"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	authv1 "github.com/openshift/api/authorization/v1"
+	projv1 "github.com/openshift/api/project/v1"
+	"github.com/openshift/origin/pkg/cmd/monitor_command"
+	"github.com/openshift/origin/pkg/monitor/intervalcreation"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	monitorserialization "github.com/openshift/origin/pkg/monitor/serialization"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	namespace            = "openshift-e2e-disruption-monitor"
+	serviceAccountName   = "disruption-monitor-sa"
+	varLogPath           = "/var/log"
+	disruptionDataFolder = "disruption-data"
+	disruptionTypeEnvVar = "DISRUPTION_TYPE_LABEL"
+)
+
+var disruptionDataPath = fmt.Sprintf("%s/%s", varLogPath, disruptionDataFolder)
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("apiserver")
+
+	g.It("start in-cluster disruption monitors [Early]", func() {
+		ctx := context.Background()
+		createTestBed(ctx, oc)
+	})
+
+	g.It("tear down in-cluster disruption monitors [Late]", func() {
+		ctx := context.Background()
+		deleteTestBed(ctx, oc)
+
+		nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		var events monitorapi.Intervals
+		for _, node := range nodes.Items {
+			nodeEvents, err := fetchNodeInClusterEvents(ctx, &node)
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
What happens with Expect here if one node has an issue, and others don't? I'd like to clearly log the problem but continue, I'd rather have intervals from all the other hosts than lose more than we need to.

Reply:
Good idea, updated the code to collect errors and print a warning instead of throwing error

Reply:
Reword to collect errors and show warning instead of failing the test